transformer_engine/common/fused_attn/fused_attn_f16_max512_seqlen.cu

```diff
@@ -1235,15 +1235,21 @@ void fused_attn_max_512_fwd_qkvpacked(
 
     void *devPtrS = nullptr;
 
-    if (Aux_Output_Tensors->size == 0) {
-        Aux_Output_Tensors->size = 1;
-        Tensor *output_S = reinterpret_cast<Tensor *>(Aux_Output_Tensors->tensors[0]);
+    if (Aux_CTX_Tensors->size == 0) {
+        Aux_CTX_Tensors->size = 2;
+        Tensor *output_S = reinterpret_cast<Tensor *>(Aux_CTX_Tensors->tensors[0]);
         output_S->data.dptr = nullptr;
         output_S->data.shape = {batch, num_head, max_seqlen, max_seqlen};
         output_S->data.dtype = input_QKV->data.dtype;
-    } else if (Aux_Output_Tensors->size == 1) {
-        Tensor *output_S = reinterpret_cast<Tensor *>(Aux_Output_Tensors->tensors[0]);
+        Tensor *output_rng_state = reinterpret_cast<Tensor *>(Aux_CTX_Tensors->tensors[1]);
```

Comment:
Yes, you don't have to change anything on the fmha_v1 front. I did this because I thought when you finish PR 227, you might need the rng_state. Now that 227 is merged, are you saying that rng_state is still not needed? Is it because the dropout information is packed in the softmax tensor `output_S` as its sign bits?

Reply:
Yes, unlike the flash attention, the dropout information is packed in the softmax tensor, so fmha_v1 doesn't need the rng_state in backward

Reply:
Ok, in that case, I'll remove rng_state from the m512 backward.

Reply:
I only removed the read of `out_rng_state` in the `else if (Aux_Output_Tensors->size == 2)` case, but not the write of `out_rng_state` in the `if (Aux_Output_Tensors->size == 0)` case. The PyTorch modules are still outputting `rng_state` as part of its aux ctx tensors, same for the `f16_arbitrary` and `fp8` cases. I can make `f16_m512` an exception but I think keeping this here is probably cleaner.